Firestore/core/src/firebase/firestore/model/field_value.h

```diff
@@ -225,6 +220,24 @@ class FieldValue {
   };
 };
 
+// TODO(rsgowman): Expand this to roughly match the java class
+// c.g.f.f.model.value.ObjectValue. Probably move it to a similar namespace as
+// well. (FieldValue itself is also in the value package in java.) Also do the
+// same with the other FooValue values that FieldValue can return.
+struct ObjectValue {
```

Comment:
Do we still need this struct, considering the code here is in final stage (i.e. no further change to the map type)?

Reply:
Sort of. The next step is to rework this type as a wrapper around a FV that contains a map. (This allows us to avoid passing generic FV's everywhere and then immediately asserting that they're actually a map.) That'll be the next PR (already in progress).